packages/map-template/src/components/Search/Search.jsx

```diff
@@ -517,52 +553,70 @@ function Search({ onSetSize, isOpen }) {
                 </label>
             </div>
 
-            { /* Horizontal list of Categories */}
-
-            {searchResults.length > 0 && selectedCategory && (<div className="search__back">
-                <button type="button" className="search__back-button" onClick={handleBack}>
-                    <ChevronLeft />
-                </button>
-                <div className="search__back-text">
-                    {categories?.find(([category]) => category === selectedCategory)[1]?.displayName}
-                </div>
-            </div>
+            {/* Vertical list of Categories */}
+            {/* Show full category list only when searchResults are empty */}
+            {isInputFieldInFocus && !showNotFoundMessage && categories.length > 0 && searchResults.length === 0 && (
+                <Categories
+                    onSetSize={onSetSize}
+                    searchFieldRef={searchFieldRef}
+                    getFilteredLocations={(category) => getFilteredLocations(category)}
+                    isOpen={!!selectedCategory}
+                />
             )}
-            {isInputFieldInFocus && !showNotFoundMessage && categories.length > 0 && searchResults.length === 0 && <Categories onSetSize={onSetSize}
-                searchFieldRef={searchFieldRef}
-                getFilteredLocations={category => getFilteredLocations(category)}
-                isOpen={!!selectedCategory}
-            />}
-
-            { /* Message shown if no search results were found */}
 
+            {/* Message shown if no search results were found */}
             {showNotFoundMessage && <p className="search__error"> {t('Nothing was found')}</p>}
 
-
-
-            { /* Vertical list of search results. Scrollable. */}
-
-            {searchResults.length > 0 &&
+            {/* When search results are found (category is selected or search term is used) */}
+            {searchResults.length > 0 && (
                 <div className="search__results prevent-scroll" {...scrollableContentSwipePrevent}>
-                    {searchResults.map(location =>
-                        <ListItemLocation
-                            key={location.id}
-                            location={location}
-                            locationClicked={() => onLocationClicked(location)}
-                            isHovered={location?.id === hoveredLocation?.id}
-                        />
-                    )}
-                </div>
-            }
 
+                    {/* If category is selected, render back button, sub-categories (if any) and Locations for that parent category */}
+                    {selectedCategory && (
+                        <>
+                            <div className="search__nav">
```

Comment:
"search__nav" could be its own BEM Block, "search-nav". Then we can also use proper BEM Elements for the button and text.